src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
```suggestion
public interface MachineProducer<? extends Machine> {
```
just T gives us possibility to do even so: SomeProducer implements MachineProducer String (String in diamond operator. )